engine/src/main/java/com/arcadedb/query/sql/parser/ContainsTextCondition.java

```diff
@@ -39,11 +39,11 @@ public ContainsTextCondition(SqlParser p, int id) {
   @Override
   public boolean evaluate(Identifiable currentRecord, CommandContext ctx) {
     Object leftValue = left.execute(currentRecord, ctx);
-    if (leftValue == null || !(leftValue instanceof String)) {
```

Comment:
Let's see : 

A. ```(x==null || !(x instanceof Y))```

B. ```(!(x instance of Y))```

If x is NULL , then A is TRUE, B is also TRUE because [null is a sub-class of nothing](https://stackoverflow.com/questions/2950319/is-null-check-needed-before-calling-instanceof)

If x is something and is subclass of Y, then both A and B are false

if x is something that is not a subclass of Y, then both A and B are true


So, it turns out A and B are equivalent for all sorts of x  values, hence the proposed simplification.

Reply:
You're right, changing it.